recirq/quantum_chess/quantum_board.py

```diff
@@ -728,31 +728,35 @@ def post_select_on(
         return result
 
     def do_move(self, m: move.Move) -> int:
-        """Will call do_move_internal to perform a move on the quantum board,
-        resetting the quantum circuit if the board is in a fully classical position.
-        """
-        initial_circuit = self.circuit.copy()
-        move_succeeded = self.do_move_internal(m)
-        if self.reset_starting_states and initial_circuit != self.circuit:
-            self.reset_if_classical()
-        return move_succeeded
-
-    def do_move_internal(self, m: move.Move) -> int:
         """Performs a move on the quantum board.
         Based on the type and variant of the move requested,
         this function augments the circuit, classical registers,
-        and post-selection criteria to perform the board.
+        and post-selection criteria to perform the move. The
+        circuit may be cleared if reset_if_classical is enabled and running the
+        circuit (which occurs only with that option enabled) shows that
+        the board is in a classical state.
+
         Returns:  The measurement that was performed, or 1 if
             no measurement was required.
         """
-        if not m.move_type:
-            raise ValueError("No Move defined")
-        if m.move_type == enums.MoveType.NULL_TYPE:
-            raise ValueError("Move has null type")
-        if m.move_type == enums.MoveType.UNSPECIFIED_STANDARD:
-            raise ValueError("Move type is unspecified")
-
-        # Reset accumulations here because function has conditional return branches
+        move_func = {
+            enums.MoveType.JUMP: self._do_move_jump,
```

Comment:
It would be nice if we could make this a constant somehow.  It seems like this is basically a static dictionary if not for the "self.".

Reply:
Done.